lib/ruby_lsp/global_state.rb

```diff
@@ -1,27 +1,38 @@
 # typed: strict
 # frozen_string_literal: true
 
-require "singleton"
-
 module RubyLsp
-  class DependencyDetector
-    include Singleton
+  class GlobalState
```

Comment:
We could use [`Singleton`](https://docs.ruby-lang.org/en/3.3/Singleton.html) to prevent instantiation but we'd need a workaround for tests that are creating new ones.

Reply:
We're trying to move away from using `Singleton` as it makes everything accessible globally.